### DISCUSSION TITLE
What is the meaning of mask in aggregation? can we use filterNode to instead?

### DISCUSSION THREAD

**Discussion by yx-keith** [trinodb/trino#23889] [Q&A] [answered]
I don't understand mask in aggregation, does anyone understand it?

**Comment by martint** [comment, marked as answer]
It's used to implement filtered aggregations: https://trino.io/docs/current/functions/aggregate.html#filtering-during-aggregation

The aggregation operator uses the mask column to determine whether to include the corresponding input in the aggregation function computation.

**Reply by yx-keith** [reply]
thank you very much martint, I get it

**Reply by yx-keith** [reply]
in this case, can we add a filterNode before aggregationNode in optimization phase?  is it easier to understand?

**Reply by yx-keith** [reply]
it's a little hard to understand the mask
